### PR TITLE
Step initial version and E2E tests

### PR DESCRIPTION
**Pull request by imrekel**

### CODE REVIEW

File: e2e/get_access_token.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -ex
+
+json_response=$(curl -X POST https://auth.services.bitrise.dev/auth/realms/bitrise-services/protocol/openid-connect/token -k \
+    --data "client_id=artifact-pull" \
+    --data "client_secret=$ARTIFACT_PULL_AUTH_CLIENT_SECRET" \
+    --data "grant_type=urn:ietf:params:oauth:grant-type:uma-ticket" \
+    --data "scope=build_artifact:read build:read app:read" \
+    --data "claim_token=eyJidWlsZF9pZHMiOlsiNzNkMzNmYjUtMzVjNi00OTVmLWJkODAtMDE1YWU2ODFkYjMzIiwiMzk0MDRiZWUtNTJiYS00Y2EyLTg1MDgtOTE0ODllN2Y2YWZhIiwiZWQwZGEwY2YtNjZjYy00MTA5LWIyM2YtOGExNTZkNjFiMGMzIiwiMDUxMzBjZTQtODI1Yi00Y2ExLWE5YmUtNGY1NDQxM2U1ZGNkIl0sInBpcGVsaW5lX2lkIjpbIjM2ZTg1NDBkLTQxYzctNDNjZS05NzRiLTgyNTQ5OTc2OTkxZSJdfQ==" \
```
Comment (shams-ahmed) [inline e2e/get_access_token.sh:10]: is this a valid token?

**Review by shams-ahmed — APPROVED** [review]
Look good, just one comment about the token

File: e2e/get_access_token.sh (hunk 1 above)
Comment (sylank) [reply e2e/get_access_token.sh:10]: It is a base64 token, it contains a build slugs array. 3rd party users can't generate new tokens without the `ARTIFACT_PULL_AUTH_CLIENT_SECRET`.

Comment (imrekel) [reply e2e/get_access_token.sh:10]: it's not an actual token just a base64 encoded json

**Review by sylank — APPROVED** [review]
LGTM 👍

**Pull request merged**